starter-code/js/index.js

```diff
@@ -1,11 +1,32 @@
-var $cart = document.querySelector('#cart tbody');
-var $calc = document.getElementById('calc');
+var $cart = document.querySelector("#cart tbody");
+var $calc = document.getElementById("calc");
+var $delete = document.querySelector(".btn.btn-delete");
 
 function updateSubtot($product) {
   // Iteration 1.1
+  let basePrice = $product.querySelector(".pu > span").innerHTML;
+  let qty = $product.querySelector(".qty > label > input").value;
+  let subTotal = basePrice * qty;
+  $product.querySelector(".subtot > span").innerHTML = subTotal;
+  return subTotal;
 }
 
 function calcAll() {
   // Iteration 1.2
+  let total = 0;
+  document.querySelectorAll(".product").forEach(function($product) {
+    total += updateSubtot($product);
+  });
+  document.querySelector("h2 > span").innerHTML = total;
 }
-$calc.onclick = calcAll;
\ No newline at end of file
+
+$calc.onclick = calcAll;
+
+function deleteOnClick(element) {
+  element.querySelector(".rm > button").onclick = function() {
+    $cart.removeChild(element);
+  };
+}
+document.querySelectorAll("#cart > tbody > tr").forEach(ele => {
```

Comment:
This right here will create an error if you would ever do the "add new product" task. The reason for this is that this line right here will never be updated => To solve this, I would have put line 30-32 within your deleteOnClick function and then you can just call this deleteOnClick whenever you add a new product.